### PR TITLE
PHC-4329 - Update TrackTile tests

### PR DESCRIPTION
**Pull request by namick**
## Changes

  - Fix and enable 12 test files in components/TrackTile (still working on 5)
  - Update mocking in jest.setup.ts
  - Update @react-native-picker/picker mocking

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -9,7 +9,11 @@ module.exports = {
   moduleFileExtensions: ['js', 'json', 'ts', 'tsx', 'jsx', 'node', 'mjs'],
   modulePathIgnorePatterns: [
     '<rootDir>[/\\\\](|coverage|node_modules)[/\\\\]',
-    'src/components/TrackTile'
+    'src/components/TrackTile/ManageTrackers/ManageTrackers.test.tsx',
+    'src/components/TrackTile/TrackerDetails/UnitPicker.test.tsx',
+    'src/components/TrackTile/TrackerDetails/AdvancedTrackerEditor/AdvancedTrackerEditor.test.tsx',
+    'src/components/TrackTile/TrackerDetails/AdvancedTrackerDetails/AdvancedTrackerDetails.test.tsx',
+    'src/components/TrackTile/TrackerDetails/TrackerDetails.test.tsx'
```
Comment (namick) [inline jest.config.js:16]: Enables 12 test files, these 5 still have issues.

File: src/components/TrackTile/TrackerRow/TrackerRow.tsx [hunk 2]
```diff
@@ -45,12 +45,12 @@ export const TrackerRow: FC<TrackerRowProps> = (props) => {
...
       )}
       {!loading &&
-        trackers.map((tracker) => (
+        trackers.map((tracker, i) => (
           <TouchableOpacity
             testID={tID(
               `open-tracker-details-button-${tracker.metricId || tracker.id}`,
             )}
-            key={tracker.metricId}
+            key={i}
```
Comment (namick) [inline src/components/TrackTile/TrackerRow/TrackerRow.tsx:53]: Fixes an error in tests

File: src/components/TrackTile/hooks/test/useRecentCodedValues.test.ts [hunk 3]
```diff
@@ -43,11 +43,13 @@ describe('useTrackerValues', () => {
...
-    const { result, waitFor } = renderHook(() =>
+    const { result, waitForNextUpdate } = renderHook(() =>
       useRecentCodedValues('metricId'),
     );
 
-    await waitFor(() => expect(result.current).toEqual([]));
+    await waitForNextUpdate();
+
+    expect(result.current).toEqual([]);
   });
```
Comment (namick) [inline src/components/TrackTile/hooks/test/useRecentCodedValues.test.ts:53]: Fixes this "not wrapped in act" error.

Screenshot 2023-03-24 at 8 55 04 PM

**Review by markdlv — APPROVED** [review]
Still curious about the `@react-native-picker/picker` swap, but approving since that change/discussion is in a separate PR.

**Pull request merged**